website/src/pages/tutorial.md

```diff
@@ -0,0 +1,523 @@
+Note the trick we pulled with `{ for details }`. `Option<_>` implements `Iterator` so we can use it to display the only
+element returned by the `Iterator` with the `{ for ... }` syntax.
+
+### Handling state
+
+Remember the `use_state` used earlier? That is a special function, called a "hook". Hooks are used to "hook" into 
+lifecycle of a function component and perform actions. You can learn more about this hook, and others 
+[here](/next/concepts/function-components/pre-defined-hooks#use_state)
+
+:::note
+Struct components act differently. See the documentation to learn about those.
+:::
+
+## Fetching data (using external REST API)
+
+In a real world application, data will basically never be hardcoded but instead will come from an API. Let's fetch our 
+videos list from external source. For this we will need to add the following crates:
+- [`reqwasm`](https://crates.io/crates/reqwasm)
+  For making the fetch call.
+- [`serde`](https://serde.rs) with derive features
+  For de-serializing the JSON response 
+- [`wasm-bindgen-futures`](https://crates.io/crates/wasm-bindgen-futures)
+  For executing Rust Future as a Promise
+
+Let's update the dependencies in `Cargo.toml` file:
+
+```toml title="Cargo.toml"
+[dependencies]
+reqwasm = "0.2"
+serde = { version = "1.0", features = ["derive"] }
+wasm-bindgen-futures = "0.4"
+```
+
+Update the `Video` struct to derive the `Deserialize` trait:
+
+```rust {1-2}
+- #[derive(Clone, PartialEq)]
++ #[derive(Clone, PartialEq, Deserialize)]
+struct Video {
+    id: usize,
+    title: String,
+    speaker: String,
+    url: String,
+}
+```
+
+Now as the last step, we need to update our `App` component to make the fetch request instead of using hardcoded data
+
+```rust {3-23,32-33}
+#[function_component(App)]
+fn app() -> Html {
+-    let videos = vec![
+-        // ...
+-    ]
++    let videos = use_state(|| vec![]);
++    {
++        let videos = videos.clone();
++        use_effect(move || {
++            let videos = videos.clone();
++            wasm_bindgen_futures::spawn_local(async move {
++                let fetched_videos: Vec<Video> = Request::get("https://yew.rs/tutorial/data.json")
++                    .send()
++                    .await
++                    .unwrap()
++                    .json()
++                    .await
++                    .unwrap();
++                videos.set(fetched_videos);
++            });
++            || ()
++        });
++    }
+
+    // ...
+ 
+    html! {
+        <>
+            <h1>{ "RustConf Explorer" }</h1>
+            <div>
+                <h3>{"Videos to watch"}</h3>
+-                <VideosList videos={videos} on_click={on_video_select.clone()} />
++                <VideosList videos={(*videos).clone()} on_click={on_video_select.clone()} />
+            </div>
+            { for details }
+        </>
+    }
+}
+```
+
+:::note
+We're using `unwrap`s here because it is a demo application. In a real world, you want to have proper error handling.
```

Comment:
```We're using `unwrap`s here because it is a demo application. In a real world, you want to have [proper error handling](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html).```